iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java

```diff
@@ -279,6 +332,51 @@ public static Anonymizer write(Network n, Path xmlFile) {
         return write(n, new ExportOptions(), xmlFile);
     }
 
+    public static Anonymizer write(Network network, ExportOptions options, DataSource dataSource) throws IOException {
+        try (OutputStream osb = dataSource.newOutputStream(null, XIIDM, false);
+             BufferedOutputStream bosb = new BufferedOutputStream(osb)) {
+
+            if (options.getMode() == IidmImportExportMode.NO_SEPARATED_FILE_FOR_EXTENSIONS) {
```

Comment:
I don't understand why anonymization is written before writing the network and why is done only in case of no separation.